crates/wasmparser/src/primitives.rs

```diff
@@ -195,16 +195,36 @@ pub struct ResizableLimits {
     pub maximum: Option<u32>,
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct ResizableLimits64 {
+    pub initial: u64,
+    pub maximum: Option<u64>,
+}
+
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub struct TableType {
     pub element_type: Type,
     pub limits: ResizableLimits,
 }
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
-pub struct MemoryType {
-    pub limits: ResizableLimits,
-    pub shared: bool,
+pub enum MemoryType {
+    B32 {
```

Comment:
What is "B" here? Not that it matters much, but I guess I would expect "Memory" or simply "M".

Reply:
I thought "bits", but I like your thinking of "Memory" more!